src/app/api/chat/route.ts

```diff
@@ -16,7 +16,7 @@ export async function POST(req: Request) {
     const lastMessage = messages[messages.length - 1]
 
     // Get the context from the last message
-    const context = await getContext(lastMessage.content, '')
+    const context = await getContext(typeof lastMessage.content == 'string' ? lastMessage.content : lastMessage.content[0].text, '')
```

Comment:
for JS/TS we typically use strict comparison `===` unless we expect to be able to type convert for comparison such as `'1' ==1`

```suggestion
    const context = await getContext(typeof lastMessage.content === 'string' ? lastMessage.content : lastMessage.content[0].text, '')
```